hail/src/main/scala/is/hail/io/TypedCodecSpec.scala

```diff
@@ -34,6 +34,12 @@ final case class TypedCodecSpec(_eType: EType, _vType: Type, _bufferSpec: Buffer
     (rt, (in: InputStream) => new CompiledDecoder(_bufferSpec.buildInputBuffer(in), f))
   }
 
+  def buildStructDecoder(requestedType: TStruct): (PStruct, (InputStream) => Decoder) = {
```

Comment:
I'm unreasonably afraid of that syntax because it fails if `pt` is `null` (in Java, `null instanceof T` is false for any T). Upon reflection, buildDecoder should never return `null`, so I'll change.